Cargo.toml

```diff
@@ -67,7 +67,7 @@ log = "^0.4"
 mockito = "^1"
 murmur3 = "0.5.2"
 once_cell = "1"
-opendal = "0.47"
+opendal = "0.48"
```

Comment:
Nit: I think it would be better to do this in a separate PR

Reply:
Got, let me upgrade it first.

Reply:
This PR relies new APIs introduced by 0.48.

Reply:
Hi, @liurenjie1024 and @Fokko, please review and merge https://github.com/apache/iceberg-rust/pull/500 first.